src/tree/SubscriptionTreeItem.ts

```diff
@@ -21,46 +21,65 @@ import { GroupTreeItemBase } from './GroupTreeItemBase';
 import { ResourceGroupTreeItem } from './ResourceGroupTreeItem';
 
 export class SubscriptionTreeItem extends SubscriptionTreeItemBase {
+
+    private async getResourceGroups(context: IActionContext): Promise<ResourceGroup[]> {
+        if (!this.cache.resourceGroups.length || !this._keepCache) {
+            const client: ResourceManagementClient = await createResourceClient([context, this.subscription]);
+            this.cache.resourceGroups = await uiUtils.listAllIterator(client.resourceGroups.list());
+        }
+        return this.cache.resourceGroups;
+    }
+
     public readonly childTypeLabel: string = localize('resourceGroup', 'Resource Group');
 
-    private _nextLink: string | undefined;
-    private _items: GroupableResource[] = [];
+    private cache: {
+        resourceGroups: ResourceGroup[];
+        nextLink?: string;
+        appResources: AppResource[];
+    }
     private _treeMap: { [key: string]: GroupTreeItemBase } = {};
 
-    private rgsItem: AppResource[] = [];
+    private resetCache(): void {
+        this.cache = {
+            resourceGroups: [],
+            appResources: [],
+        }
+    }
+
+    private _keepCache: boolean = false;
 
     public constructor(parent: AzExtParentTreeItem, subscription: ISubscriptionContext) {
         super(parent, subscription);
-        this.registerRefreshEvents('groupBy')
+        this.resetCache();
+        this.registerRefreshEvents('groupBy');
     }
 
     public hasMoreChildrenImpl(): boolean {
-        return !!this._nextLink;
+        return !!this.cache.nextLink;
     }
 
     public async loadMoreChildrenImpl(clearCache: boolean, context: IActionContext): Promise<AzExtTreeItem[]> {
-        if (clearCache) {
-            this._nextLink = undefined;
-            this.rgsItem = [];
+        if (clearCache && !this._keepCache) {
```

Comment:
I'm a little uncomfortable having two separate flags for this.  `clearCache` is handled internally by `AzExtParentTreeItem`.  I think you should be able to utilize `clearCache` in `loadMoreChildrenImpl` to do everything that you need.

If you really want to keep your own flag though, make sure you always put the state changes in a `try/finally`.  If there's ever an error (for example, if `this.createTreeMaps` ended with an error) we'd never change the state to false.

We could also consider exposing `treeProvider.refreshUIOnly` command as well.

Reply:
> We could also consider exposing `treeProvider.refreshUIOnly` command as well.

I think that's a good idea. There's a lot of use cases for it.

Reply:
I removed the additional flag, but at the cost of not caching as much as before. The clearCache flag is not aggressive enough (or at least we need to expose a way for us to easily call refresh without setting clearCache to true. 

I don't think `treeProvider.refreshUIOnly` will help in this case, because calling it does not result in `loadMoreChildrenImpl` getting called.